### ISSUE TITLE
move_motors / moveMotors has unpredictable results.

### ISSUE THREAD

**Issue by rhfogh** [mxcube/HardwareRepository#129]
The issue has arisen in branch 2.1, but is likely more general. 

When you call a function like MultiCollect.move_motors(motor_position_dict) or
Diffractometer.moveMotors(motor_position_dict) you assume that this will move the motors to the positions given in the input dictionary. Unfortunately this is not generally the case. That is a specific problem that I need a fix for.

This is probably a special case of a more general problem. I am in an unusual position in that I am not coding primarily for a few, specific beamlines but for the entirety of MXCuBE. This means I cannot tailor my work to a specific  target but have to rely on high level functions with predictable results. Furthermore I am doing things that are apparently not that common, such as setting kappa, kappa_phi and translation motors programmatically, which may bring out problems that have otherwise been hidden. Which raises the question of what general effects a function like move_motors is supposed to have across the entire collaboration, and how you can make sure that this is not broken at individual sites. 

The first and smallest problem is in SampleCentringQueueEntry.execute(), which calls self.diffractometer_hwobj.moveMotors(). On ESRF ID30B it turned out that 1) the motors were moved in random order, depending on the dictionary passed in. 2) moving the kappa motor had the side effect of changing the position of the translational motors to unpredictable values. It looks like this may be caused by the MicrodiffKappaMotor.getNewSamplePosition() function, that calculates new translation positions to match the new kappa value. The particular problem could be relatively easily fixed by moving kappa and kappa_phi first, and the other motors with a second function call. But it leaves the risk that similar problems may be (or become) present in other installations.

The bigger problem is in AbstractMultiCollect.do_collect(), which calls self.move_motors(motors_to_move_before_collect) in order to position the sample correctly for data collection. After testing it turned out that on ESRF ID30B (and, it seems ID29) the ID30BMultiCollect.move_motors(motors_to_move_dict) functions explicitly ignored kappa and kappa_phi input values. That would not be particularly hard for me to undo, but there is presumably a reason why this feature was introduced, and I could hardly change the behaviour of other people's beamlines. Alternatively I would either have to reimplement and bypass the do_collect function, or to code specifically for each individual beamline. Either approach would be laborious and hard to maintain. 

So, summing up:
How can I pass motor positions reliably to the data collection?
And what can we do to ensure a set of reliable high-level functions across different sites?
